File: Dart/File/read_file_and_return_as_bytes.dart
import "dart:core";
import "dart:io";

Future<void> main() async {
  // Return the content of a file in bytes.
  final List<int> bytesFromFile = await readFileAndReturnAsBytes("assets/valid/j3U5uJY779L49q98MX86iFsxs2kY9ew3/README.md");
  print(bytesFromFile);
}

// Return the contents of a file as bytes after reading it.
Future<List<int>> readFileAndReturnAsBytes(final String path) async {
  return File(path).readAsBytes();
}

File: Go/File/read-file-and-return-as-bytes.go
package main

import (
	"io"
	"log"
	"os"
)

func main() {
	// Read a file and than return the content as bytes.
	readFileAndReturnAsBytes("assets/valid/valid-json.json")
}

// Read a file and than return the content as bytes
func readFileAndReturnAsBytes(path string) []byte {
	file, err := os.Open(path)
	if err != nil {
		log.Fatalln(err)
	}
	content, err := io.ReadAll(file)
	if err != nil {
		log.Fatalln(err)
	}
	err = file.Close()
	if err != nil {
		log.Fatalln(err)
	}
	return content
}
